src/fights.ts

```diff
@@ -1066,7 +1146,42 @@ const freeFightSources = [
 
   new FreeFight(
     () => (have($familiar`Machine Elf`) ? clamp(5 - get("_machineTunnelsAdv"), 0, 5) : 0),
-    () => adv1($location`The Deep Machine Tunnels`, -1, ""),
+    () => {
+      propertyManager.setChoices({
+        1119: 5, //escape DMT
+      });
+      if (saleValue($item`abstraction: certainty`) >= saleValue($item`abstraction: thought`)) {
+        acquire(1, $item`abstraction: thought`, saleValue($item`abstraction: certainty`), false);
+      }
+      if (saleValue($item`abstraction: joy`) >= saleValue($item`abstraction: action`)) {
+        acquire(1, $item`abstraction: action`, saleValue($item`abstraction: joy`), false);
+      }
+      if (saleValue($item`abstraction: motion`) >= saleValue($item`abstraction: sensation`)) {
+        acquire(1, $item`abstraction: sensation`, saleValue($item`abstraction: motion`), false);
+      }
+      adventureMacro(
+        $location`The Deep Machine Tunnels`,
+        Macro.externalIf(
+          saleValue($item`abstraction: certainty`) >= saleValue($item`abstraction: thought`),
```

Comment:
may be worth naming these 3 bools given that they show up twice each. May also not be worth that!